T8105: Modify workflows failed due to pullrequest_target change

asklymenko opened the pull request:
## Change summary
<!--- Provide a general summary of your changes in the Title above -->
The pull_request_target workflows are now evaluated against the base repository’s default branch for branch protections and environment rules, not the pull request’s target branch.

It is required to change the structure to have only 1 common file for all branches. This workflow should call a branch-specific workflow to run tests. In these changes I use a local workflow of a branch (the branch is not specified in the workflow path). According to the GitHub documentation it should call a workflow of the same branch, if a local workflow specified. Let's try this approach to fix current problem.

## Types of changes
<!---
What types of changes does your code introduce? Put an 'x' in all the boxes that apply.
NOTE: Markdown requires no leading or trailing whitespace inside the [ ] for checking
the box, please use [x]
-->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)
<!-- optional: Link to related other tasks on Phabricator. -->
* https://vyos.dev/TT8105

## Related PR(s)
<!-- Link here any PRs in other repositories that are required by this PR -->
* https://github.com/vyos/.github/pull/107

## How to test / Smoketest result
<!---
Please describe in detail how you tested your changes. Include details of your testing
environment, and the tests you ran. When pasting configs, logs, shell output, backtraces,
and other large chunks of text, surround this text with triple backtics
```
like this
```

Or provide the output of the smoketest

```
$ /usr/libexec/vyos/tests/smoke/cli/test_xxx_feature.py
test_01_simple_options (__main__.TestFeature.test_01_simple_options) ... ok
```
-->

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
<!--- The entire development process is outlined here: https://docs.vyos.io/en/latest/contributing/development.html -->
- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [ ] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

asklymenko committed:
T8105: Modify workflows failed due to pullrequest_target change

It is required to change this workflow to be called by another workflow - trigger-package-smoketest.yml.

asklymenko committed:
T8105: Modify workflows failed due to pullrequest_target change

Use one file for all branches to trigger the package-smoketest.yml workflow.

asklymenko committed:
T8105: Modify workflows failed due to pullrequest_target change

Specify the workflow location in "./package-smoketest.yml" format to show that it is a local workflow.

asklymenko committed:
T8105: Modify workflows failed due to pullrequest_target change

Use full path for local workflows

asklymenko committed:
T8105: Modify workflows failed due to pullrequest_target change

Removed conditional smoketest jobs for specific branches and simplified to a single smoketest job. No need to specify a branch when local workflow is called.

asklymenko commented:
@kumvijaya Could you please review this approach?

kumvijaya reviewed (approved):
Can approve as this needs quick-fix.
But reusable workflow to be kept in .github repo (not on same repo)

asklymenko commented:
@kumvijaya 
Thank you. If it is better to keep reusable workflows in .github repo I will move it to that repo later.

asklymenko committed:
T8105: Modify workflows failed due to pullrequest_target change

Change this workflow to trigger the reusable workflow from "vyos/.github" repo.

asklymenko committed:
T8105: Modify workflows failed due to pullrequest_target change

Delete .github/workflows/package-smoketest.yml. It was moved to another repository. It will be used as a reusable workflow from that repo.

asklymenko committed:
T8105: Modify workflows failed due to pullrequest_target change

Change the workflow to run the branch-specific reusable workflow.

c-po reviewed (commented):
The package smoketest workflow is ONLY required in vyos-1x repo - is it really necessary to make it an overcomplicated re-usable workflow?

kumvijaya commented:
> The package smoketest workflow is ONLY required in vyos-1x repo - is it really necessary to make it an overcomplicated re-usable workflow?

Agree. If it is used only in this repo, can avoid creating reusable workflow.

asklymenko commented:
@c-po @kumvijaya I think we should have the same logic for other branches, in this case it is ok to move that file to other repo. To have the same location for all repos.

c-po commented:
> @c-po @kumvijaya I think we should have the same logic for other branches, in this case it is ok to move that file to other repo. To have the same location for all repos.

I strongly disagree as the logic becomes overly complex for no reason.

asklymenko commented:
I am closing this PR, this issue will be resolved by [another PR](https://github.com/vyos/vyos-1x/pull/4914).
